crates/storage/db/benches/criterion.rs

```diff
@@ -0,0 +1,134 @@
+use criterion::{criterion_group, criterion_main, Criterion};
+use reth_db::{
+    cursor::DbCursorRW,
+    database::Database,
+    mdbx::{test_utils::create_test_rw_db, WriteMap},
+    table::*,
+    tables::*,
+    transaction::DbTxMut,
+};
+
+/// Returns bench vectors in the format: `Vec<(Key, EncodedKey, Value, CompressedValue)>`.
+///
+/// TBD, so for now only loads 3 default values.
+fn load_vectors<T>() -> Vec<(T::Key, bytes::Bytes, T::Value, bytes::Bytes)>
+where
+    T: Table + Default,
+    T::Key: Default + Clone,
+    T::Value: Default + Clone,
+{
+    let encoded: bytes::Bytes = bytes::Bytes::copy_from_slice(T::Key::default().encode().as_ref());
+    let compressed: bytes::Bytes =
+        bytes::Bytes::copy_from_slice(T::Value::default().compress().as_ref());
+
+    let row = (T::Key::default(), encoded, T::Value::default(), compressed);
+    vec![row.clone(), row.clone(), row]
+}
+
+macro_rules! impl_criterion {
```

Comment:
Nit but should we make these generic instead of with macro? And we can pass string params for the test name

Reply:
yeah true, changed it for `criterion`.

However,  `iai` api is pretty basic and doesn't really allow for that, as far as i can see.